src/app/upload_gmail/page.tsx

```diff
@@ -16,15 +18,49 @@ export default function Page() {
 	const { data: session, status, update } = useSession()
 	const [log, setLog] = React.useState<LogRecord[]>([]);
 	const [uploadedPairs, setUploadedPairs] = React.useState<Set<string>>(new Set());
+	const [queuePairs, setQueuePairs] = React.useState<Set<string>>(new Set());
 	const [addedPairs, setAddedPairs] = React.useState<number>(0);
 	const [nextPageToken, setNextPageToken] = React.useState<string>('');
 	const [gmailQuery, setGmailQuery] = React.useState<string>('');
 	const [processedMessages, setProcessedMessages] = React.useState<number>(0);
 	const [totalMessages, setTotalMessages] = React.useState<number | null>(null);
-	const [startDate, setStartDate] = React.useState<string>('');
-	const [endDate, setEndDate] = React.useState<string>('');
+	const [startDate, setStartDate] = React.useState<Date | null>(null);
+	const [endDate, setEndDate] = React.useState<Date | null>(null);
 	const [domain, setDomain] = React.useState<string>('');
 
+  type Errors = {
+	start?: string;
+	end?: string;
+  };
+
+  const validate = () => {
+	const errs: Errors = {};
+	if (!startDate) {
+	  errs.start = "Start date is required";
```

Comment:
these error messages are never accessed,. it just checks if there are > 0 errors, and then ignore whatever the content was. maybe they were intended to be logged?